DashSync/Models/Chain Managers/DSTransactionManager.m

```diff
@@ -0,0 +1,639 @@
+                    [self.chain setBlockHeight:TX_UNCONFIRMED andTimestamp:0 forTxHashes:@[hash]];
+                }
+            }
+        }
+    }
+    
+    if (notify) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (rescan) {
+                [[DSEventManager sharedEventManager] saveEvent:@"transaction_manager:tx_rejected_rescan"];
+                UIAlertController * alert = [UIAlertController
+                                             alertControllerWithTitle:DSLocalizedString(@"transaction rejected", nil)
+                                             message:DSLocalizedString(@"Your wallet may be out of sync.\n"
+                                                                       "This can often be fixed by rescanning the blockchain.", nil)
+                                             preferredStyle:UIAlertControllerStyleAlert];
+                UIAlertAction* cancelButton = [UIAlertAction
+                                               actionWithTitle:DSLocalizedString(@"cancel", nil)
+                                               style:UIAlertActionStyleCancel
+                                               handler:^(UIAlertAction * action) {
+                                               }];
+                UIAlertAction* rescanButton = [UIAlertAction
+                                               actionWithTitle:DSLocalizedString(@"rescan", nil)
+                                               style:UIAlertActionStyleDefault
+                                               handler:^(UIAlertAction * action) {
+                                                   [self.chainManager rescan];
+                                               }];
+                [alert addAction:cancelButton];
+                [alert addAction:rescanButton];
+                [[[[UIApplication sharedApplication] keyWindow] rootViewController] presentViewController:alert animated:YES completion:nil];
+                
+            }
+            else {
+                [[DSEventManager sharedEventManager] saveEvent:@"transaction_manager_tx_rejected"];
+                UIAlertController * alert = [UIAlertController
+                                             alertControllerWithTitle:DSLocalizedString(@"transaction rejected", nil)
+                                             message:@""
+                                             preferredStyle:UIAlertControllerStyleAlert];
+                UIAlertAction* okButton = [UIAlertAction
+                                           actionWithTitle:DSLocalizedString(@"ok", nil)
+                                           style:UIAlertActionStyleCancel
+                                           handler:^(UIAlertAction * action) {
+                                           }];
+                [alert addAction:okButton];
+                [[[[UIApplication sharedApplication] keyWindow] rootViewController] presentViewController:alert animated:YES completion:nil];
```

Comment:
`rootViewController` is not reliable, it's better to get the current top controller:
https://github.com/dashevo/dashsync-iOS/blob/master/DashSync/Models/DSAuthenticationManager.m#L225